lldb/source/Core/ModuleList.cpp

```
@@ -1277,6 +1310,45 @@ static SharedModuleList &GetSharedModuleList() {
   return GetSharedModuleListInfo().module_list;
 }
 
+static std::shared_ptr<llvm::cas::ObjectStore>
+GetOrCreateCASStorage(FileSpec CandidateConfigSearchPath) {
+  auto &shared_module_list = GetSharedModuleListInfo();
+  if (shared_module_list.cas_object_store)
+    return shared_module_list.cas_object_store;
+
+  std::scoped_lock<std::mutex> lock(shared_module_list.shared_lock);
+  // Config CAS from properties.
+  llvm::cas::CASConfiguration cas_config;
+  cas_config.CASPath =
+      ModuleList::GetGlobalModuleListProperties().GetCASOnDiskPath().GetPath();
+  cas_config.PluginPath =
+      ModuleList::GetGlobalModuleListProperties().GetCASPluginPath().GetPath();
+  cas_config.PluginOptions =
+      ModuleList::GetGlobalModuleListProperties().GetCASPluginOptions();
+
+  if (!cas_config.CASPath.empty()) {
+    if (auto maybe_cas = cas_config.createDatabases()) {
+      shared_module_list.cas_object_store = std::move(maybe_cas->first);
+      return shared_module_list.cas_object_store;
+    } else
+      llvm::consumeError(maybe_cas.takeError());
+  }
+
+  // Try search from candidiate path.
+  auto search_config = llvm::cas::CASConfiguration::createFromSearchConfigFile(
+      CandidateConfigSearchPath.GetPath());
+  if (!search_config)
+    return nullptr;
+
+  if (auto maybe_cas = search_config->second.createDatabases()) {
+    shared_module_list.cas_object_store = std::move(maybe_cas->first);
+    return shared_module_list.cas_object_store;
+  } else
+    llvm::consumeError(maybe_cas.takeError());
```

We should probably bubble this error up so we can give a precise failure reason. Maybe joinError() it with a more descriptive higher-level error at a layer above.